---
title: "QC On Raw Data"
author: "Stephen Pederson"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.height = 7,
  fig.width = 10
)
```

```{r packages}
library(ngsReports)
library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(plotly)
library(msa)
```

```{r options}
panderOptions("table.split.table", Inf)
panderOptions("big.mark", ",")
theme_set(theme_bw())
```

```{r config}
config <- here::here("config/config.yml") %>%
  read_yaml()
suffix <- paste0(config$tags$tag, config$ext)
sp <- config$ref$species %>%
  str_replace("(^[a-z])[a-z]*_([a-z]+)", "\\1\\2") %>%
  str_to_title()
```

```{r samples}
samples <- config$samples %>%
  here::here() %>%
  read_tsv() %>%
  mutate(
    R1 = paste0(sample, config$tags$r1, suffix),
    R2 = paste0(sample, config$tags$r2, suffix),
  ) %>%
  pivot_longer(
    cols = c("R1", "R2"),
    names_to = "Reads",
    values_to = "Filename"
  ) %>%
  mutate_if(
    function(x){length(unique(x)) < length(x)},
    as.factor
  )
```

```{r check_config}
config$analysis <- config$analysis %>%
  lapply(intersect, y = colnames(samples)) %>%
  .[vapply(., length, integer(1)) > 0]
```

```{r add_columns}
if (length(config$analysis)) {
  samples <- samples %>%
    unite(
      col = group, 
      any_of(as.character(unlist(config$analysis))), 
      sep = "_", remove = FALSE
    )
} else {
  samples$group <- samples$Filename
}
```


```{r group_cols}
group_cols <- hcl.colors(
  n = length(unique(samples$group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(samples$group))
```


```{r setFigureSizes}
fh <- round(6 + nrow(samples) / 7, 0)
```

# Quality Assessment on Raw Data {.tabset}

```{r rawFqc}
rawFqc <- here::here("data/raw/FastQC") %>%
  list.files(pattern = "fastqc.zip", full.names = TRUE, recursive = TRUE) %>%
  FastqcDataList() %>%
  .[fqName(.) %in% samples$Filename]
```

```{r changeRawFqcPath}
for (i in seq_along(rawFqc)){
  run <- str_extract(path(rawFqc[[i]]), "Hiseq_[12]")
  rawFqc[[i]]@Summary$Filename <- paste(run, rawFqc[[i]]@Summary$Filename, sep = "/")
} 
```

## FastQC Summary

```{r plotSummary, fig.height=fh, fig.cap = "*Overall summary of FastQC reports*"}
plotSummary(rawFqc)
```

## Library Sizes

A total of `r length(rawFqc)` libraries were contained in this dataset, with read totals ranging between `r pander(comma(range(readTotals(rawFqc)$Total_Sequences)))` reads.

Across all libraries, reads were between `r pander(range(as.numeric(getModule(rawFqc, "Sequence_Length")$Length)))` bases.
This does indicate some read trimming had been performed prior to that undertaken here.

```{r plotReadTotals, fig.height=fh, fig.cap = "*Library Sizes for all supplied fastq files. Any samples run as multiple libraries are  shown as the supplied multiple libraries and have not been merged.*"}
plotReadTotals(rawFqc, pattern = suffix, usePlotly = TRUE)
```

## Sequence Quality

```{r plotBaseQuals, fig.height=fh, fig.cap = "*Mean sequencing quality scores at each base position for each library*"}
plotBaseQuals(
  rawFqc,
  pattern = suffix, 
  usePlotly = TRUE,
  dendrogram = TRUE,
  cluster = TRUE
  )
```


## GC Content

GC content patterns for both R1 and R2 reads were highly reminiscent of those seen when rRNA has not been completely removed.

```{r plotR1GC, fig.cap="*GC content for all R1 libraries*"}
plotGcContent(rawFqc[grepl("R1", fqName(rawFqc))], plotType = "line", usePlotly = TRUE, gcType = "Trans") 
```

```{r plotR2GC, fig.cap="*GC content for all R2 libraries*"}
plotGcContent(rawFqc[grepl("R2", fqName(rawFqc))], plotType = "line", usePlotly = TRUE, gcType = "Trans") 
```


## Sequence Content

```{r plotSeqContentResid, fig.cap = "*Base and Position specific residuals for each sample. The mean base content at each position was calculated for each nucleotide, and the sample-specific residuals calculated.*"}
plotly::ggplotly(
  getModule(rawFqc, module = "Per_base_sequence_content") %>% 
    mutate(Base = fct_inorder(Base)) %>%
    group_by(Base) %>% 
    mutate(
      across(c("A", "C", "G", "T"), function(x){x - mean(x)}) 
    ) %>% 
    pivot_longer(
      cols = c("A", "C", "G", "T"), 
      names_to = "Nuc", 
      values_to = "resid"
    ) %>%
    separate(Filename, into = c("Run", "Filename"), sep = "/") %>%
    left_join(samples) %>%
    unite(Filename, Run, Filename, sep = "/") %>%
    ggplot(
      aes(Base, resid, group = Filename, colour = group)
    ) + 
    geom_line() +
    facet_wrap(~Nuc) + 
    scale_colour_manual(values = group_cols) +
    labs(
      x = "Read Position", y = "Residual", colour = "Group"
    )
)
```

## AdapterContent

```{r plotAdapterContent, fig.height=fh, fig.cap="*Total Adapter Content for each sample shown by starting position in the read.*"}
plotAdapterContent(
  x = rawFqc, 
  pattern = suffix, 
  usePlotly = TRUE,
  dendrogram = TRUE,
  cluster = TRUE
  )
```

## Overrepresented Sequences

```{r os}
os <- suppressMessages(getModule(rawFqc, "Over"))
os_fh <- min(20, 6 + nrow(os) / 20)
```

```{r plotOverRep, fig.height=os_fh, fig.cap = "*Summary of over-represented sequences across all libraries*"}
if (nrow(os)){
  if (length(unique(os$Filename)) > 1){
    suppressMessages(
      plotOverrep(
        x = rawFqc,
        pattern = suffix, 
        usePlotly = TRUE,
        dendrogram = TRUE,
        cluster = TRUE
      )
    )
  }
}
```

```{r}
os %>%
  group_by(Sequence, Possible_Source) %>%
  summarise(
    `Found in` = n(),
    Total = sum(Count),
    `Largest Percent` = glue("{round(max(Percentage), 2)}%")
  ) %>%
  ungroup() %>%
  arrange(desc(Total)) %>%
  dplyr::filter(Total > 5e5) %>%
  mutate(Rank = seq_along(Total)) %>%
  pander(
    caption = "*Summary of over-represented sequences within the raw data.*"
  )
```

```{r}
os %>%
  group_by(Sequence, Possible_Source) %>%
  summarise(
    `Found in` = n(),
    Total = sum(Count),
    `Largest Percent` = glue("{round(max(Percentage), 2)}%")
  ) %>%
  ungroup() %>%
  arrange(desc(Total)) %>%
  pull("Sequence") %>%
  setNames(paste0("os", seq_along(.))) %>%
  .[c(1:3, 7, 8, 10, 11, 12, 18, 19)] %>%
  as("DNAStringSet") %>%
  msa() %>%
  print(show = 'complete')
```



A BLAST search of 10 manually selected sequences (which are clearly the same sequence) revealed this to be a perfect match to multiple genomic locations such as [RPS29](https://www.ncbi.nlm.nih.gov/nucleotide/NG_050638.2), [RN7SL3](https://www.ncbi.nlm.nih.gov/nucleotide/NR_145670.1), [NEMF](https://www.ncbi.nlm.nih.gov/nucleotide/NG_051075.1) and [RN7SL396P](https://www.ncbi.nlm.nih.gov/nucleotide/NG_042962.1).
It is clear that this is likely to have some association with rRNA.


```{r}
os %>%
  group_by(Sequence, Possible_Source) %>%
  summarise(
    `Found in` = n(),
    Total = sum(Count),
    `Largest Percent` = glue("{round(max(Percentage), 2)}%")
  ) %>%
  ungroup() %>%
  arrange(desc(Total)) %>%
  pull("Sequence") %>%
  setNames(paste0("os", seq_along(.))) %>%
  .[(c(9, 13, 17))] %>%
  as("DNAStringSet") %>%
  msa() %>%
  print(show = 'complete')
```

A manual search using additional over-represented sequences which did not align to the above consensus, revealed matches to additional SRP RNAs and [7S RNA](https://www.ncbi.nlm.nih.gov/nucleotide/V00477.1)

As such, rRNA may present some issues for this dataset as these sequences will not be removed by adapter removal. 
Alignment to a rRNA genome may be of benefit.

